### PR TITLE
Add events for creating and damaging bastions

### PR DESCRIPTION
**Pull request by CivFactions**
Added cancellable events that fire when a bastion is created or damaged
Also updated pom to Java 1.8 and spigot 1.10.2

### CODE REVIEW

File: Bastion/pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
      <dependency>
            <groupId>org.spigotmc</groupId>
            <artifactId>Spigot</artifactId>
-           <version>1.10</version>
```
Comment (ProgrammerDan) [inline Bastion/pom.xml]: This is a custom version of spigot (ugh, legacy reasons) and is already 1.10.2 behind the scenes.

File: Bastion/src/isaac/bastion/manager/BastionBlockManager.java [hunk 2]
```diff
@@ -38,13 +41,11 @@
...
 import vg.civcraft.mc.citadel.Citadel;
 import vg.civcraft.mc.citadel.reinforcement.PlayerReinforcement;
-import vg.civcraft.mc.namelayer.NameAPI;
-import vg.civcraft.mc.namelayer.NameLayerPlugin;
 import vg.civcraft.mc.namelayer.permission.PermissionType;
 
 public class BastionBlockManager {
 	public BastionBlockSet set;
-	private Map<String, Long> playerLastEroded = new HashMap<String, Long>();
+	private Map<Player, Long> playerLastEroded = new HashMap<Player, Long>();
```
Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/manager/BastionBlockManager.java]: General advice is not to directly cache Player objects, but to cache UUID and retrieve Player object at-point-of-need.

Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/manager/BastionBlockManager.java]: This is why in other plugins you'll see Map<UUID, X> pattern used extensively, etc.

**Comment by ProgrammerDan** [comment]
Other then the caching of Player objects directly and the unneeded Spigot version change due to behind the scenes build server trickery, this looks good.

Oh, and besides that you forgot to add the actual event objects in your commit :P

**Comment by CivFactions** [comment]
lol. That's what I get for trying to do this quick before leaving work.

File: Bastion/src/isaac/bastion/event/BastionDamageEvent.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+
+import isaac.bastion.BastionBlock;
+
+public class BastionDamageEvent extends PlayerEvent implements Cancellable {
+	public enum Cause { BLOCK_PLACED, PEARL }
```
Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/event/BastionDamageEvent.java]: Add ELYTRA as well, i forget if I merged that code upstream yet or not but it's either in this fork or Devoted's.

Comment (CivFactions) [inline Bastion/src/isaac/bastion/event/BastionDamageEvent.java]: added

**Comment by CivFactions** [comment]
ok @ProgrammerDan I've tested this version on my local dev server.

**Pull request merged**